datastores/gossip_kv/server/simulation.rs

```diff
@@ -0,0 +1,430 @@
+//! network partitions.
+//!
+//! ## Overview
+//!
+//! Conceptually, the simulation contains a "Fleet", which is a collection of "Hosts". These
+//! aren't real hosts, but rather a collection of individual Hydroflow transducers (one per host)
+//! that can communicate with each other over a virtual network. Every host has a "hostname"
+//! which uniquely identifies it within the fleet.
+//!
+//! ```text
+//!  ┌───────────────────────────────────────────────────────────────────────────────────────────┐
+//!  │SIMULATION                                                                                 │
+//!  │ ┌───────────────────────────────────────────────────────────────────────────────────────┐ │
+//!  │ │FLEET                                                                                  │ │
+//!  │ │ ┌───────────────────────────────┐                   ┌───────────────────────────────┐ │ │
+//!  │ │ │HOST                           │                   │HOST                           │ │ │
+//!  │ │ │ ┌──────┐   ┌──────┐  ┌──────┐ │                   │ ┌──────┐   ┌──────┐  ┌──────┐ │ │ │
+//!  │ │ │ │INBOX │   │INBOX │  │INBOX │ │                 ┌-┼-►INBOX │   │INBOX │  │INBOX │ │ │ │
+//!  │ │ │ └──┬───┘   └──┬───┘  └──┬───┘ │                 │ │ └──┬───┘   └──┬───┘  └──┬───┘ │ │ │
+//!  │ │ │ ┌──▼──────────▼─────────▼───┐ │                 │ │ ┌──▼──────────▼─────────▼───┐ │ │ │
+//!  │ │ │ │                           │ │                 │ │ │                           │ │ │ │
+//!  │ │ │ │        TRANSDUCER         │ │                 │ │ │        TRANSDUCER         │ │ │ │
+//!  │ │ │ │                           │ │                 │ │ │                           │ │ │ │
+//!  │ │ │ └───┬─────────┬──────────┬──┘ │                 │ │ └───┬─────────┬─────────┬───┘ │ │ │
+//!  │ │ │  ┌──▼───┐  ┌──▼───┐  ┌───▼──┐ │                 │ │  ┌──▼───┐  ┌──▼───┐  ┌──▼───┐ │ │ │
+//!  │ │ │  │OUTBOX│  │OUTBOX│  │OUTBOX┼-┼--┐              │ │  │OUTBOX│  │OUTBOX│  │OUTBOX│ │ │ │
+//!  │ │ │  └──────┘  └──────┘  └──────┘ │  │              │ │  └──────┘  └──────┘  └──────┘ │ │ │
+//!  │ │ └───────────────────────────────┘  │              │ └───────────────────────────────┘ │ │
+//!  │ └────────────────────────────────────┼──────────────┼───────────────────────────────────┘ │
+//!  │                                    ┌─┼──────────────┼─┐                                   │
+//!  │                                    │ └--------------┘ │                                   │
+//!  │                                    │ NETWORK MESSAGE  │                                   │
+//!  │                                    │    PROCESSING    │                                   │
+//!  │                                    └──────────────────┘                                   │
+//!  └───────────────────────────────────────────────────────────────────────────────────────────┘
+//! ```
```

Comment:
niceee

Reply:
I guess we will want to make this module a feature of hydroflow eventually?

Reply:
https://github.com/hydro-project/hydroflow/pull/1416